blazor/dropdown-list/value-binding.md

```diff
@@ -0,0 +1,216 @@
+
+## Enum binding
+
+You can bind enum data to DropDownList component. The following code helps you get a string value from the enumeration data.
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/value-binding/enum-binding.razor %}
+
+{% endhighlight %}
+
+![Blazor DropDownList with Enum Data](./images/value-binding/blazor_dropdown_enum-binding.png)
+
+## Show or hide clear button
+
+You can use [ShowClearButton](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.DropDowns.SfDropDownList-2.html#Syncfusion_Blazor_DropDowns_SfDropDownList_2_ShowClearButton) property to specify whether to show or hide the clear button. When the clear button is clicked, `Value`, `Text`, and `Index` properties are reset to null.
+
+> If the TValue is non nullable type, then while using clear button, it will set the default value of the data type and TValue is set as nullable type while using clear button it will set to null value(eg: If TValue is int while clearing 0 is set to the component and if TValue is int? while clearing null is set to the component)
+
+The below sample demonstrates `string` is used as `TValue`. So, if we clear the value using clear button, it will set to null as it's the default value of the respective type.
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/value-binding/show-hide-clear-button.razor %}
+
+{% endhighlight %}
+
+![Blazor DropDownList with clear button](./images/value-binding/blazor_dropdown_show-hide-clear-button.png)
+
+## Dynamically change TItem
+
+You can change the `TItem` property dynamically which defines the datasource type of the DropDownList component with help of `@typeparam` directive. The below sample demonstration explains how  to change  the TItem dynamically with different type of datasource.
+
+### Creating generic dropdownList component
+
+First, you need to create a `DropDownList.razor` file in the `/Pages` folder as a parent component. Also add a Parameter property for a List as `<TItem>` and `TValue`.
+
+{% tabs %}
+{% highlight razor %}
+
+@using Syncfusion.Blazor.DropDowns;
+@typeparam TValue;
+@typeparam TItem;
+
+<SfDropDownList TValue="TValue" Width="300px" TItem="TItem" @bind-Value="@DDLValue" Placeholder="Please select a value" DataSource="@customData">
+    <DropDownListFieldSettings Text="Text" Value="ID"></DropDownListFieldSettings>
+</SfDropDownList>
+
+@code {
+    [Parameter]
+    public List<TItem> customData { get; set; }
+    [Parameter]
+    public TValue DDLValue { get; set; }
+    [Parameter]
+    public EventCallback<TValue> DDLValueChanged { get; set; }
+}
+{% endtabs %}
+{% endhighlight razor %}
+
+### Usage of generic component with different type
+
+Then you can render the Generic DropDownList component with required TValue and TItem in the respective razor components. 
+
+Here, we have rendered the DropDownList component with TValue as string type in the `/Index.razor` file and DropDownList component with TValue as int? type in the `/Counter.razor` file.

```

Comment:
Here, the DropDownList component is rendered with the TValue as a string type in the `/Index.razor` file and the DropDownList component with TValue as an int type in the `/Counter.razor` file.